src/haxelib/api/LibraryData.hx

```diff
@@ -0,0 +1,99 @@
+package haxelib.api;
+
+import haxe.DynamicAccess;
+
+import haxelib.api.Vcs.VcsID;
+
+/** Exception thrown upon errors regarding library data, such as invalid versions. **/
+class LibraryDataException extends haxe.Exception {}
+
+/**
+	Library version, which can be used in commands.
+
+	This type of library version has a physical folder in the project root directory
+	(i.e. it is not a dev version)
+**/
+abstract Version(String) to String from SemVer from VcsID {
+	inline function new(s:String) {
+		this = s;
+	}
+
+	public static function ofString(s:String):Version {
+		if (!isValid(s))
+			throw new LibraryDataException('`$s` is not a valid library version');
+		return new Version(s);
+	}
+
+	/** Returns whether `s` constitues a valid library version. **/
+	public static function isValid(s:String):Bool {
+		return s == Git || s == Hg || SemVer.isValid(s);
```

Comment:
This should call `VcsID.isVcs` instead of implementing it again.

Reply:
Fixed it, and renamed isVcs to isValid for consistency.